### DISCUSSION TITLE
Release planning: v0.8.0

### DISCUSSION THREAD

**Discussion by jplatte** [tokio-rs/axum#3052] [General]
There's a milestone here: https://github.com/tokio-rs/axum/milestone/7

I'd like to talk about what makes sense to be in there, and what doesn't, so we can hopefully ship a release soon-ish.

cc @mladedav @yanns @SabrinaJewson

**Comment by jplatte** [comment]
My personal take on the current open issues in the list:

- https://github.com/tokio-rs/axum/issues/2287 - nice to have, though I don't think I'm gonna pick it up myself. Does somebody else want to?
- https://github.com/tokio-rs/axum/issues/2298 - nice to have, but maybe delay it to ship a bit faster? I worked on it a bit today.
- https://github.com/tokio-rs/axum/issues/2651 - nice to have, I just pushed a PR for it that should not take long to review
- https://github.com/tokio-rs/axum/issues/2659 - nice to have, but also hard so unless somebody has a good idea soon I'd prefer to postpone
- https://github.com/tokio-rs/axum/issues/2678 - nice to have, but probably not all that important?
- https://github.com/tokio-rs/axum/pull/2941 - would really like to have it, probably the most requested out of the bunch (though still a niche thing); would be nice if folks could chime in re. the latest review
- https://github.com/tokio-rs/axum/issues/2998 - 🤷🏼

**Reply by yanns** [reply]
About https://github.com/tokio-rs/axum/pull/2941, there are still comments that you have not addressed right?

About https://github.com/tokio-rs/axum/issues/2998, it does not seem that there's a strong consensus, or that someone is willing to implement the middleware solution. I'm ok to remove it from the list.

**Reply by jplatte** [reply]
- #2941 - yes, would like to know whether other people agree with the review comments before putting more work into it
- #2998 - right, I'll move it to a new 0.9 milestone (can always move it back if others disagree)

**Reply by mladedav** [reply]
I'll try to look at #2678 and maybe we can make it in time. I agree it's not that important though so if we have everything else, we don't need to block on it.

Sorry if my review of the generic listeners demotivated you. That definitely wasn't the intention. I'd be happy to either talk the comments through or try to help with the implementation.

**Reply by jplatte** [reply]
No need to apologize, I just wasn't entirely convinced by your arguments, but couldn't write a good counterargument either, and was generally quite busy. Your reviews are much appreciated!

**Reply by yanns** [reply]
About
>     * [`IntoResponse` implementation for `(StatusCode, R) where R: IntoResponse` overrides error status codes and might be a footgun #2287](https://github.com/tokio-rs/axum/issues/2287) - nice to have, though I don't think I'm gonna pick it up myself. Does somebody else want to?

I've looked into https://github.com/tokio-rs/axum/pull/2468.
I'm ready to spend more time on it if we first decide that this is the right solution.

**Reply by jplatte** [reply]
It's maybe a little bit of a "tacked-on" solution, but I still think it would be an improvement over the status quo. So I'm still in favor. Only reason it may be worth delaying is that 0.8.0 is already going to be a rather big release, so maybe it might be better to have subtle changes like that separated out in the upgrading process?

**Reply by yanns** [reply]
I'm ok for it. Should we remove https://github.com/tokio-rs/axum/issues/2287 from the 0.8 milestone then?

**Reply by jplatte** [reply]
If you don't care about it much, sure (edit: done).

**Comment by yanns** [comment]
About https://github.com/tokio-rs/axum/pull/2941, I took a look, but I don't feel qualified enough to assess the comments and the possible alternatives.

**Reply by jplatte** [reply]
Thanks for looking anyways. On second thought I did like mladedav's suggestions, so have implemented them (w/ slight adjustments).
